CMakeLists.txt

```diff
@@ -127,16 +127,16 @@ target_compile_features(DDC INTERFACE cxx_std_17)
 target_include_directories(
     DDC
     INTERFACE
+        "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/generated>"
         "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
         "$<INSTALL_INTERFACE:include>"
 )
 target_link_libraries(DDC INTERFACE Kokkos::kokkos)
 target_compile_definitions(DDC INTERFACE MDSPAN_USE_PAREN_OPERATOR=1)
```

Comment:
About `target_compile_definitions(DDC INTERFACE MDSPAN_USE_PAREN_OPERATOR=1)`

I think we should rework DDC to be able to adapt whether it is enabled or not.

Reply:
see #677